docs/library/graphing/plotly.md

```diff
@@ -10,104 +10,144 @@ import reflex as rx
-        "padding": 5,
-        "width": "100%",
-        "overflow_x": "auto",
-    }
-}
 ```
+## 3D graphing example
+Let's create a 3D surface plot of Mount Bruno. 
 
-Plotly is a graphing library that can be used to create interactive graphs.
+```python demo exec
+import plotly.graph_objects as go
+import pandas as pd
 
-Let's create a line graph of life expectancy in Canada as an example.
+# Read data from a csv
+z_data = pd.read_csv('https://raw.githubusercontent.com/plotly/datasets/master/api_docs/mt_bruno_elevation.csv')
 
-```python eval
-rx.center(
-    rx.plotly(data=line()),
-    height="400px",
-    style=style["box"],
+fig = go.Figure(data=[go.Surface(z=z_data.values)])
+fig.update_traces(contours_z=dict(show=True, usecolormap=True,
+                                  highlightcolor="limegreen", project_z=True))
+fig.update_layout(
+    scene_camera_eye=dict(x=1.87, y=0.88, z=-0.64),
+    width=500, height=500,
+    margin=dict(l=65, r=50, b=65, t=90)
 )
+
+def mountain_surface():
+    return rx.center(
+        rx.plotly(data=fig),
+    )
 ```
 
-First create a plotly figure. In this example we use plotly express to do so.
+## Plot as State Var
+Set plot as a State var when the plot need to be updated during run time. 
 
-```python
+```python demo exec
 import plotly.express as px
+class PlotlyState(rx.State):
+    df = px.data.gapminder().query(f"country=='Canada'")
+    figure = px.line(
+        df,
+        x="year",
+        y="lifeExp",
+        title="Life expectancy in Canada",
+    )
 
-df = px.data.gapminder().query("country=='Canada'")
-fig = px.line(df, x="year", y="lifeExp", title='Life expectancy in Canada')      
-```
+    def set_selected_country(self, country):
+        self.df = px.data.gapminder().query(f"country=='{country}'")
+        self.figure = px.line(
+            self.df,
+            x="year",
+            y="lifeExp",
+            title=f"Life expectancy in {country}",
+        )
 
-Now pass the plotly figure to the Plotly component.
 
-```python
-rx.plotly(data=fig, height="400px")
+def line_chart_with_state():
+    return rx.vstack(
+        rx.select(
+            ['China', 'France', 'United Kingdom', 'United States', 'Canada'],
+            default_value="United States",
+            on_change=PlotlyState.set_selected_country,
+        ),
+        rx.plotly(data=PlotlyState.figure),
+    )
 ```
 
-Now lets take a look at a more complex example.
-Let's create a 3D surface plot of Mount Bruno.
+## Adding Styles and Layouts
+Use `update_layout()` method to update the layout of your chart. Checkout [Plotly Layouts](https://plotly.com/python/reference/layout/) for all layouts props. 
 
-```python eval
-rx.center(
-  rx.vstack(
-      rx.plotly(data=mount()),
-  ),
-  height="400px",
-  style=style["box"],
-)
+```md alert info
+Note that the width and height props are not recommended to ensure the plot remains size responsive to its container. 
```

Comment:
It is not clear what these width and height props are relating to in this example, is it the update_layout or is it rx.plotly